bob-access/src/credentials.rs

```diff
@@ -64,8 +65,20 @@ impl Credentials {
         CredentialsBuilder::default()
     }
 
-    pub fn ip(&self) -> Option<IpAddr> {
-        Some(self.address?.ip())
+    pub fn ip(&self) -> &Option<Vec<SocketAddr>> {
+        &self.address
+    }
+
+    pub fn set_addresses(&mut self, addresses: Vec<SocketAddr>) {
+        self.address = Some(addresses);
+    }
+
+    pub fn single_ip(&self) -> Option<SocketAddr> {
```

Comment:
You must either use all IP addresses or store only one inside this struct. Storing all IPs and use only one is not consistent